### PR TITLE
ss-5: Add BaseButton component

### PR DESCRIPTION
**Pull request by Anatolii-Petrenko**
This issue depends on Issue #2 "Common styles"

### CODE REVIEW

File: frontend/src/bundles/common/components/button/button.tsx [hunk 1]
```diff
@@ -1,10 +1,44 @@
+import React from 'react';
+
+import styles from './styles.module.scss';
+
 type Properties = {
-    label: string;
+    children: React.ReactNode;
     type?: 'button' | 'submit';
+    className?: string;
+    variant?: 'primary' | 'secondary' | 'plain' | 'round';
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/button/button.tsx:9]: Consider adding enum for such cases

Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/components/button/button.tsx]: Done in commit [47118be](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/47118be99a52c78ea88273915c6e45684e5815c4)

File: frontend/src/bundles/common/components/button/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    --secondary-font-weight: var(--font-weight-1-button-text);
+    --secondary-text-color: #01cacc;
+    --secondary-hover-text-color: #0be9ee;
+    --plain-text-color: var(--brand-color-pink-100);
+    --plain-hover-text-color: var(--background-blue-600);
+    --plain-font-size: var(--font-size-2-button-text);
+    --plain-font-weight: var(--font-weight-2-button-text);
+    --round-button-color: var(--brand-color-pink-100);
+    --round-button-dimension: 28px;
+}
```
[the remark this answers is not in the document]
Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/components/button/styles.module.scss]: Done in [1610174](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/1610174ddf544e441d599c9b62e890f48237168e)

File: frontend/src/bundles/common/enums/button.enum.ts
[the remark this answers is not in the document]
Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/enums/button.enum.ts:1]: Done in [1610174](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/1610174ddf544e441d599c9b62e890f48237168e)

**Comment by Anatolii-Petrenko** [comment]
Refactoring according to the comment is done.
By passing different properties to the component, such buttons could be obtained:
Screenshot 2023-03-15 at 15 54 55

File: frontend/src/bundles/common/components/button/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,134 @@
+@import url("../../../../assets/css/variables.css");
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/button/styles.module.scss:1]: you do not need this import. we load all styles in the root component

File: frontend/src/bundles/common/components/button/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    --primary-color: var(--color-white-100);
+    --small-font-size: 16px;
+    --small-font-weight: var(--font-weight-400);
+    --secondary-text-color: #01cacc;
+    --secondary-hover-text-color: #0be9ee;
+    --plain-text-color: var(--color-pink-100);
+    --plain-hover-text-color: var(--color-blue-600);
+    --round-button-color: var(--color-pink-100);
+    --round-button-dimension: 28px;
+}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/button/styles.module.scss:30]: Not sure why you need this

File: frontend/src/bundles/common/pages/base.tsx [hunk 5]
```diff
@@ -1,7 +1,161 @@
 import React from 'react';
 
+import { Button } from '../components/components';
+// import { ButtonSize, ButtonVariant } from '../enums/enums';
+import { ButtonSize } from '../enums/button-size.enum';
+// import { ButtonType } from '../enums/button-type.enum';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/pages/base.tsx:6]:
```suggestion
```

File: frontend/src/bundles/common/pages/base.tsx [hunk 6]
```diff
@@ -1,7 +1,161 @@
 import React from 'react';
 
+import { Button } from '../components/components';
+// import { ButtonSize, ButtonVariant } from '../enums/enums';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/pages/base.tsx:4]:
```suggestion
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/button/styles.module.scss (hunk 4 above)
Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/components/button/styles.module.scss]: Removed unnecessary variables in [a14e6a4](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/a14e6a4d4d44118fcf3c291f456d982a528d7a7b)

File: frontend/src/bundles/common/pages/base.tsx (hunk 5 above)
Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/pages/base.tsx]: Removed in [a14e6a4](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/a14e6a4d4d44118fcf3c291f456d982a528d7a7b)

File: frontend/src/bundles/common/pages/base.tsx (hunk 6 above)
Comment (Anatolii-Petrenko) [reply frontend/src/bundles/common/pages/base.tsx]: Removed in [a14e6a4](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/21/commits/a14e6a4d4d44118fcf3c291f456d982a528d7a7b)

**Comment by Anatolii-Petrenko** [comment]
This issue depends on [ss-7: Add Base Modal #19](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/19) because I use "--px-2" variable added in this issue.

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**